src/apps/chifra/pkg/articulate/log.go

```diff
@@ -0,0 +1,51 @@
+package articulate
+
+import (
+	"encoding/hex"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/abi"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+)
+
+func ArticulateLog(log *types.SimpleLog, abiMap abi.AbiInterfaceMap) (articulated *types.SimpleFunction, err error) {
```

Comment:
Don't do it now, but at some point, I feel like this will have an interface thus:

```
func (log *types.SimpleLog) Articulate(abiMap abi.AbiInterfaceMap) (articulated *types.SimpleFunction, err error)
```

In other words `logs`, `transactions` and `traces` will implement an interface called `Articulate`.